### DISCUSSION TITLE
Rendering Images on iOS and Android with fo-dicom 5.0.3

### DISCUSSION THREAD

**Discussion by christopher-schmidt-dedalus** [fo-dicom/fo-dicom#1460] [Q&A] [answered]
Hi Community,

How can I render the DicomImage in iOS and Android in fo-dicom 5.0.3? At the moment I am using fo-dicom in combination with Xamarin.
In version 4.0.x I was using the `.As<T>();` method and it works fine. Because of some other troubles in the lib (for example NetworkException) I had to update.
With version 5.0.3 in iOS I try to use:
```
new DicomSetupBuilder().RegisterServices(s => s.AddFellowOakDicom().AddImageManager<ImageSharpImageManager>()).Build();
UIImage uIImage = dicomImage.RenderImage().As<UIImage>();
Stream stream = uIImage.AsPNG().AsStream();
```
And I will get die following exception:
```
FellowOakDicom.Imaging.DicomImagingException: Cannot cast to 'UIImage'; type must be assignable from 'Image`1'
  at FellowOakDicom.Imaging.ImageBase`1[TImage].As[T] () [0x00089] in <5e7598491ad24eb1adad4250322a84d7>:0 
```

Does anyone have experience with that or do I make some mistakes or is there an other way to get the image displayed?

Thx for helping

**Comment by gofal** [comment, marked as answer]
It's just as the exception tells you.
When working on other platform than windows, then I recommend to use ImageSharpImageManager to use ImageSharp (https://github.com/SixLabors/ImageSharp) instead of System.Drawing. This ImageSharp is pure .net and has no dependencies to GDI. But that means, that you have to cast to SixLabors.ImageSharp.Image after rendering, like `Image<Bgra> img = dicomImage.RenderImage().AsSharpImage()`
for futher manipulations using the Image<Bgra> class take a look at the documentation of ImageSharp.
